changeset:   259:9f9a18d15637
tag:         tip
user:        Jesse Keating <jkeating at redhat dot com>
date:        Tue Jun 19 22:19:19 2007 -0400
summary:     Create a little package sack out of dep providers to grab just the newest ones

diff -r 98bb99f7295b -r 9f9a18d15637 pypungi/gather.py
--- a/pypungi/gather.py	Wed Jun 13 18:04:20 2007 -0400
+++ b/pypungi/gather.py	Tue Jun 19 22:19:19 2007 -0400
@@ -119,7 +119,9 @@ class Gather(yum.YumBase):
                 self.logger.warn("Unresolvable dependency %s in %s.%s" % (r, po.name, po.arch))
                 continue
 
-            for dep in deps:
+            depsack = yum.packageSack.ListPackageSack(deps)
+
+            for dep in depsack.returnNewestByNameArch():
                 self.tsInfo.addInstall(dep)
                 if not self.config.has_option('default', 'quiet'):
                     self.logger.info('Added %s.%s for %s.%s' % (dep.name, dep.arch, po.name, po.arch))

